libopae/src/event.c

```diff
@@ -583,6 +583,11 @@ fpga_result __FPGA_API__ fpgaCreateEventHandle(fpga_event_handle *event_handle)
 
 fpga_result __FPGA_API__ fpgaDestroyEventHandle(fpga_event_handle *event_handle)
 {
+	//sanity check
+	if (!event_handle) {
+		return FPGA_INVALID_PARAM;
+	}
+
```

Comment:
The event handle is being passed by reference, so the dereference on 591 is actually correct.
To follow the convention that variables be declared before executable statements, please change the code to the following:

struct _fpga_event_handle *_eh;
fpga_result result = FPGA_OK;
int err = 0;

if (!event_handle) {
    return FPGA_INVALID_PARAM;
}

_eh = (struct _fpga_event_handle *) *event_handle;